cloud/spawn.go

```diff
@@ -203,7 +203,11 @@ func CreateVolume(ctx context.Context, env evergreen.Environment, volume *host.V
 // assumes distro already modified to have one region
 func CheckInstanceTypeValid(ctx context.Context, d distro.Distro, requestedType string, allowedTypes []string) error {
 	if !utility.StringSliceContains(allowedTypes, requestedType) {
-		return errors.New("This instance type has not been allowed by admins")
+		// if it's not in the settings list, check the distro
+		originalInstanceType, ok := d.ProviderSettingsList[0].Lookup("instance_type").StringValueOK()
+		if !ok || originalInstanceType != requestedType {
```

Comment:
an alternative style to the if-if could be to append the originalInstanceType to `allowedTypes` before checking `utility.StringSliceContains(allowedTypes, requestedType)`

Reply:
Yes that would also work. I guess I was thinking there was no point in having duplicates in the allowedTypes list, and there was no point in doing the provider settings lookup if the instance type was in allowedTypes anyway. But they're functionally the same.